elastic/eui
# Upgrading EUI to use Node `v18.16.0`

## Summary

Node version 16 is nearing end of life, so we're upgrading to version 18, long-term support. The latest version as of today is `v18.16.0`. PR updates the Node version used in CI, removes a legacy Webpack module for native support, and updates the Dockerfile to use a current, slim base image.

This PR closes #6791.

## QA request

Please follow these steps for a good QA run. It's assumed you have `nvm` or another means to add and use different Node versions installed locally. On your terminal:

1. Install the current Node 18 LTS `nvm install 18.16.0`
2. Use Node 18 `nvm use 18.16.0`
3. `yarn`
4. Lint and run unit tests `yarn test`
5. Start dev server `yarn start`
6. All commands should pass without incident or error
7. Review the local docs site (or PR link) for any shenanigans or missing files

### QA I ran before issuing the PR

Almost every command in our `package.json` was run. Some were not feasible because they require a current Docker image or interact with the outside environment like `release`. I do not anticipate there being any issue and will update the Docker image so these tests run in CI on the closest representation to our local dev environments.

- [x] yarn
- [x] yarn start
- [x] yarn test-docker
- [x] yarn build-docs
- [x] yarn build-pack
- [x] yarn build
- [x] yarn compile-icons
- [x] yarn compile-sass
- [x] yarn extract-i18n-strings
- [x] yarn lint
- [x] yarn lint-es
- [x] yarn lint-sass
- [x] yarn lint-sass-fix
- [x] yarn test
- [x] yarn test-unit
- [x] yarn test-cypress
- [x] yarn test-cypress-dev
- [x] yarn test-cypress-a11y
- [x] yarn combine-test-coverage
- [x] yarn start-test-server
- [x] yarn yo-component
- [x] yarn yo-doc
- [x] yarn yo-changelog
- [x] yarn storybook
- [x] yarn build-storybook

---
- [x] Spot checked the docs site locally, clicking on links, snippets, props
- [x] Ran [Hadolint](https://github.com/hadolint/hadolint) on the Dockerfile and made some suggested cleanup changes
- [x] Removed the ["docker_image"](https://github.com/elastic/eui/commit/38ab7ca4443dc04bd6f3e50cf2597914ebc6e1fc#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R9) declaration from `package.json`. Best I can tell it was added as a local var when pulling Docker images several years ago.
- [x] Added an `engines {}` declaration that errors if users try to run Node commands using a non-supported version
